cfme/common/provider_views.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+from widgetastic.widget import View, Text
+
+from cfme import BaseLoggedInPage
+from widgetastic_patternfly import Dropdown
+from widgetastic_manageiq import DetailsToolBarViewSelector
+from widgetastic_manageiq import (BreadCrumb,
+                                  SummaryTable,
+                                  Button,
+                                  TimelinesView)
+
+
+class ProviderDetailsToolBar(View):
+    """
+    represents provider toolbar and its controls
+    """
+    monitoring = Dropdown(text='Monitoring')
+    configuration = Dropdown(text='Configuration')
+    reload = Button(title='Reload Current Display')
+    policy = Dropdown(text='Policy')
+    authentication = Dropdown(text='Authentication')
+
+    @View.nested
+    class view_selector(DetailsToolBarViewSelector):  # NOQA
+        pass
+
+
+class ProviderDetailsSummaryView(View):
+    """
+    represents Details page when it is switched to Summary aka Tables view
+    """
+    properties = SummaryTable(title="Properties")
+    status = SummaryTable(title="Status")
+    relationships = SummaryTable(title="Relationships")
+    overview = SummaryTable(title="Overview")
+    smart_management = SummaryTable(title="Smart Management")
+
+
+class ProviderDetailsDashboardView(View):
+    """
+     represents Details page when it is switched to Dashboard aka Widgets view
+    """
+    # todo: need to develop this page
+    pass
+
+
+class ProviderDetailsView(BaseLoggedInPage):
+    """
+     main Details page
+    """
+    title = Text('//div[@id="main-content"]//h1')
+    breadcrumb = BreadCrumb(locator='//ol[@class="breadcrumb"]')
+
+    @View.nested
+    class toolbar(ProviderDetailsToolBar):  # NOQA
+        pass
+
+    @View.nested
+    class contents(View):  # NOQA
+        pass
```

Comment:
ah, you're right. I originally had such comments but forgot to move 
https://github.com/ManageIQ/integration_tests/pull/4387/files#diff-fa6a86dc0dc65ab876b17fc13b55f6b6R238